packages/frontend/pages/vault/[vid].tsx

```diff
@@ -342,7 +343,13 @@ const Component: React.FC = () => {
     setAction(percent > existingCollatPercent ? VaultAction.ADD_COLLATERAL : VaultAction.REMOVE_COLLATERAL)
     setCollatPercent(percent)
     const debt = await getDebtAmount(vault.shortAmount)
-    const newCollat = new BigNumber(percent).times(debt).div(100)
+    let lpCollatPercent = 0
+    // If NFT is deposited, Collateral amount from LP NFT should not be included. So target collat % - lp collat % will give actual collat to remove
+    if (lpNftId) {
+      const { collateral: uniCollat } = await getUniNFTCollatDetail(lpNftId)
+      lpCollatPercent = uniCollat.div(debt).times(100).toNumber()
+    }
+    const newCollat = new BigNumber(percent - lpCollatPercent).times(debt).div(100)
```

Comment:
Do we use BN to avoid potential calculation/conversion error?
```suggestion
    let lpCollatPercent = new BigNumber(0)
    // If NFT is deposited, Collateral amount from LP NFT should not be included. So target collat % - lp collat % will give actual collat to remove
    if (lpNftId) {
      const { collateral: uniCollat } = await getUniNFTCollatDetail(lpNftId)
      lpCollatPercent = uniCollat.div(debt).times(100)
    }
    const newCollat = new BigNumber(percent).minus(lpCollatPercent).times(debt).div(100)
```